darts/models/forecasting/lgbm.py

```diff
@@ -152,6 +155,15 @@ def fit(
             Optionally, a series or sequence of series specifying past-observed covariates for evaluation dataset
         val_future_covariates : Union[TimeSeries, Sequence[TimeSeries]]
             Optionally, a series or sequence of series specifying future-known covariates for evaluation dataset
+        categorical_past_covariates
```

Comment:
Can we move this to model construction (`__init__`)?
In `fit()` this will add more complexity and we would have different APIs for the models that support categorical and the ones that do not.
- in general we aim at making the API as unified as possible (we already have some complexity with validation series and others)

With the categorical variable names already known at fitting time, we can extract them from the input (`series`, `past_covariates`, `future_covariates`) on the fly

Reply:
Makes sense. Done.